### DISCUSSION TITLE
Regarding Script Management

### DISCUSSION THREAD

**Discussion by Cetenaru** [orestbida/cookieconsent#632] [Q&A]
Hello,

First thanks for this tool !

Not an expert at all in JS, this is not necessarily a problem, I just want to be sure I understand what I am doing/must do!

Regarding Script Management [https://cookieconsent.orestbida.com/advanced/manage-scripts.html](url)

If I understand correctly I must write the lines of the code (provided by Analytics) once between the `type=plain-text` tags?

But does the full code provided by Analytics have to be written a second time (as usual) so that it can be read if validated by the visitor, or does the tool handle this all alone ?

After one or two tests by checking the console, I see that the GET to Google was “blocked by extension” by having written the code twice. When I choose … -> save preferences

**Comment by orestbida** [comment]
@Cetenaru you shouldn't write the code twice. Simply add the required attributes to let the plugin handle the script tags:
```html
<script 
    type="text/plain" 
    data-category="analytics" 
    data-service="Google Analytics 4"
>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());
    gtag('config', 'G-XXXXXXXXXX');
</script>

<script 
    type="text/plain" 
    data-category="analytics" 
    data-service="Google Analytics 4" 
    async 
    src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"
></script>

<script 
    type="text/plain" 
    data-category="analytics" 
    data-service="!Google Analytics 4"
>
    window['ga-disable-G-XXXXXXXXXX'] = true;
</script>
```

> After one or two tests by checking the console, I see that the GET to Google was “blocked by extension” 

Check your extensions, you might have an ad blocker enabled.

**Reply by Cetenaru** [reply]
Hello,

Thanks for the answer

I just put the line below, 
`<script type="text/plain" data-category="analytics" data-service="Google Analytics" async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"></script>`

When I check online with a cookie scanner (piwik.pro), he tell, that have only one cookie before consent ! (piwik.pro)

And here below the console result : 
![consent-test](https://github.com/orestbida/cookieconsent/assets/158480905/eced38c8-bfcf-4eb6-8024-3d9822a32f6f)

So unless I'm mistaken, everything is ok. Thank for your help
